libs/features/personalization/personalization.js

```diff
@@ -271,30 +272,91 @@ function parsePlaceholders(placeholders, config, selectedVariantName = '') {
   return config;
 }
 
-function getPersonalizationVariant(manifestPath, variantNames = [], variantLabel = null) {
+export const loadEntitlements = (() => {
+  let entitlements;
+  return (async () => {
+    if (!entitlements) {
+      const { default: getUserEntitlements } = await import('../../blocks/global-navigation/utilities/getUserEntitlements.js');
+      await loadIms();
```

Comment:
I think the ims logic should be handled in `getUserEntitlements` so that all consumers of this method don't duplicate the code.
Since `getUserEntitlements` is async, it should first wait for ims to be ready, then retrieve the entitlements.

Reply:
I don't think there's a clean way to add it there.  We don't have any of the ims config data, so we'd have to pass that in.

The majority of the time (all the time?) when this is called, ims will already be loaded due to gnav or other code needing IMS already.